modules/assim.sequential/inst/restart_SDAworkflow_scripts/SDA_Workflow_LAI.R

```diff
@@ -0,0 +1,361 @@
+    }
+  ) %>% stats::setNames(date.obs)
+  
+  
+  # names = date.obs
+  # for (name in names){
+  #   for (site in names(obs.cov[[name]])){
+  #     #if we don't have any observation (diag(cov)==NA) then we remove the whole site
+  #     if(length(which(!is.na(diag(obs.cov[[name]][[site]])))) == 0){
+  #       obs.cov[[name]][[site]] <- NULL
+  #       next
+  #     }
+  #     #else we do have some records
+  #     bad = which(apply(obs.cov[[name]][[site]], 2, function(x) any(is.na(x))) == TRUE)
+  #     if (length(bad) > 0){
+  #       obs.cov[[name]][[site]] = obs.cov[[name]][[site]][,-bad]
+  #       if (is.null(dim(obs.cov[[name]][[site]]))){
+  #         obs.cov[[name]][[site]] = obs.cov[[name]][[site]][-bad]
+  #       } else {
+  #         obs.cov[[name]][[site]] = obs.cov[[name]][[site]][-bad,]
+  #       }
+  #     }
+  #   }
+  # }
+  #add start.cut to restart list
+  restart$start.cut <- lubridate::as_datetime(min(lai$calendar_date))
+  restart$start.cut <- format(restart$start.cut, "%Y-%m-%d %H:%M:%S", tz = "EST")
+  
+
+#-----------------------------------------------------------------------------------------------
+#------------------------------------------ Fixing the settings --------------------------------
+#-----------------------------------------------------------------------------------------------
+#Using the found dates to run - this will help to download mets
+settings$run$site$met.start <- as.character(met.start)
+settings$run$site$met.end <- as.character(met.end)
+
+# Setting dates in assimilation tags - This will help with preprocess split in SDA code
+settings$state.data.assimilation$start.date <-as.character(sda.start)
+sda.end <- max(names(obs.mean))
+settings$state.data.assimilation$end.date <-as.character(sda.end)
+
+# --------------------------------------------------------------------------------------------------
+#---------------------------------------------- PEcAn Workflow -------------------------------------
+# --------------------------------------------------------------------------------------------------
+#info
+settings$info$date <- paste0(format(Sys.time(), "%Y/%m/%d %H:%M:%S"))
+next.oldir <- paste0(format(Sys.time(), "%Y-%m-%d-%H-%M"))
+#Update/fix/check settings. Will only run the first time it's called, unless force=TRUE
+settings <- PEcAn.settings::prepare.settings(settings, force = TRUE)
+#settings$host$rundir <- settings$rundir
+#settings$host$outdir <- settings$modeloutdir
+#settings$host$folder <- settings$modeloutdir
+setwd(settings$outdir)
+#Write pecan.CHECKED.xml
+PEcAn.settings::write.settings(settings, outputfile = "pecan.CHECKED.xml")
+# start from scratch if no continue is passed in
+statusFile <- file.path(settings$outdir, "STATUS")
+if (length(which(commandArgs() == "--continue")) == 0 && file.exists(statusFile)) {
+  file.remove(statusFile)
+}
+
+#manually add in clim files 
+con <-PEcAn.DB::db.open(settings$database$bety)
+on.exit(db.close(con))
```

Comment:
A gotcha about on.exit: Calling it without `add = TRUE` makes this call *replace* all actions scheduled in any previous `on.exit` call, like the one on line 76 of this file. I don't know why `add=TRUE` isn't the default, but I recommend always writing it in (even for the first time `on.exit` is called in a function -- someone could always come along and add one upstream later)
 
```suggestion
on.exit(db.close(con), add = TRUE)
```